osconfig_tests/test_suites/patch/patch.go

```diff
@@ -133,39 +210,74 @@ func runExecutePatchJobTest(ctx context.Context, testCase *junitxml.TestCase, te
-			if err != nil {
-				testCase.WriteFailure("Error while fetching patch job: \n%s\n", utils.GetStatusFromError(err))
-				return
-			}
+	testCase.Logf("Creating instance with image %q", testSetup.image)
+	var metadataItems []*api.MetadataItems
+	metadataItems = append(metadataItems, testSetup.startup)
+	metadataItems = append(metadataItems, compute.BuildInstanceMetadataItem("os-config-enabled-prerelease-features", "ospatch"))
+	name := fmt.Sprintf("patch-reboot-%s-%s", path.Base(testSetup.testName), suffix)
+	inst, err := utils.CreateComputeInstance(metadataItems, client, testSetup.machineType, testSetup.image, name, testProjectConfig.TestProjectID, testProjectConfig.GetZone(), testProjectConfig.ServiceAccountEmail, testProjectConfig.ServiceAccountScopes)
+	if err != nil {
+		testCase.WriteFailure("Error creating instance: %v", utils.GetStatusFromError(err))
+		return
+	}
+	defer inst.Cleanup()
 
-			if isPatchJobFailureState(res.State) {
-				testCase.WriteFailure("Patch job '%s' completed with status %v and message '%s'", job.GetName(), res.State, job.GetErrorMessage())
-				return
-			}
+	testCase.Logf("Waiting for agent install to complete")
+	if _, err := inst.WaitForGuestAttribute("osconfig_tests/", "install_done", 5*time.Second, 5*time.Minute); err != nil {
+		testCase.WriteFailure("Error waiting for osconfig agent install: %v", err)
+		return
+	}
 
-			if res.State == osconfigpb.PatchJob_SUCCEEDED {
-				if res.InstanceDetailsSummary.GetInstancesSucceeded() < 1 {
-					testCase.WriteFailure("Patch job '%s' completed with no instances patched", job.GetName())
-				}
-				return
-			}
-		}
+	testCase.Logf("Agent installed successfully")
+
+	parent := fmt.Sprintf("projects/%s", testProjectConfig.TestProjectID)
+	osconfigClient, err := gcpclients.GetOsConfigClient(ctx)
+
+	req := &osconfigpb.ExecutePatchJobRequest{
+		Parent:      parent,
+		Description: "testing reboot patch job run",
+		Filter:      fmt.Sprintf("name=\"%s\"", name),
+		Duration:    &duration.Duration{Seconds: int64(testSetup.assertTimeout / time.Second)},
+		PatchConfig: &osconfigpb.PatchConfig{Apt: &osconfigpb.AptSettings{Type: osconfigpb.AptSettings_DIST}},
+	}
+	job, err := osconfigClient.ExecutePatchJob(ctx, req)
+
+	if err != nil {
+		testCase.WriteFailure("error while executing patch job: \n%s\n", utils.GetStatusFromError(err))
+		return
+	}
+
+	testCase.Logf("Started patch job '%s'", job.GetName())
+	if err := waitPatchJob(ctx, job, testSetup.assertTimeout); err != nil {
+		testCase.WriteFailure("Patch job '%s' error: %v", job.GetName(), err)
+	}
+
+	testCase.Logf("Checking reboot count")
+	attr, err := inst.GetGuestAttribute("osconfig_tests/", "boot_count")
+	if err != nil {
+		testCase.WriteFailure("Error retrieving boot count: %v", err)
+		return
+	}
+	num, err := strconv.Atoi(attr)
+	if err != nil {
+		testCase.WriteFailure("Error parsing boot count: %v", err)
+		return
+	}
+	if num < minBootCount || num > maxBootCount {
```

Comment:
Instead of verifying bootcount is within a range, how about grabbing the value before running the patch job and verifying that it's higher afterwards? That may be more robust if you're concerned about spurious reboots before and during the patch job.

Reply:
This function has two consumers, the first is that we did reboot (min matters, max does not)
the other tests we didnt reboot (max matters min does not)